crates/subspace-archiving/src/archiver.rs

```diff
@@ -356,23 +363,33 @@ impl Archiver {
         object_mapping: BlockObjectMapping,
         incremental: bool,
     ) -> ArchiveBlockOutcome {
+        let mut archived_segments = Vec::new();
+        let mut global_object_mapping = Vec::new();
+
+        // Since mappings are produced immediately after the block is added, we can assume that
+        // the mappings from any with_initial_state() block have already been produced.
+
         // Append new block to the buffer
         self.buffer.push_back(SegmentItem::Block {
             bytes,
             object_mapping,
         });
 
-        let mut archived_segments = Vec::new();
-        let mut object_mapping = Vec::new();
-
+        // Add completed segments and their mappings for this block.
         while let Some(segment) = self.produce_segment(incremental) {
-            object_mapping.extend(self.produce_object_mappings(&segment));
+            // Any mappings for the previous BlockContinuation or Blocks have already been
+            // produced, so we only need to produce mappings for the latest Block, BlockStart, or
+            // BlockContinuation from this block.
```

Comment:
Similarly I'm not sure how this comment is helpful, we're not matching on `Block` or `BlockStart` here. Even if this is a useful comment, the location seems arbitrary.

Reply:
Happy to delete or move it.
```suggestion
```

Reply:
Let's just delete it then, I think the internals of called functions are quite clear.